benefits/eligibility/analytics.py

```diff
@@ -2,67 +2,62 @@
 The eligibility application: analytics implementation.
 """
 
-from benefits.core import analytics as core
+from benefits.core import analytics as core, models
 
 
 class EligibilityEvent(core.Event):
     """Base analytics event for eligibility verification."""
 
-    def __init__(self, request, event_type, flow_name):
+    def __init__(self, request, event_type, flow: models.EnrollmentFlow):
         super().__init__(request, event_type)
-        # pass a (converted from string to list) flow_name to preserve analytics reporting
-        eligibility_types = [flow_name]
-        # overwrite core.Event eligibility_types
-        self.update_event_properties(eligibility_types=eligibility_types)
-        self.update_user_properties(eligibility_types=eligibility_types)
+        # overwrite core.Event enrollment flow
+        self.update_enrollment_flows(flow)
 
 
 class SelectedVerifierEvent(EligibilityEvent):
-    """Analytics event representing the user selecting an eligibility verifier."""
+    """Analytics event representing the user selecting an enrollment flow."""
 
-    def __init__(self, request, eligibility_types):
-        super().__init__(request, "selected eligibility verifier", eligibility_types)
+    def __init__(self, request, flow: models.EnrollmentFlow):
+        super().__init__(request, "selected enrollment flow", flow)
 
 
 class StartedEligibilityEvent(EligibilityEvent):
     """Analytics event representing the beginning of an eligibility verification check."""
 
-    def __init__(self, request, eligibility_types):
-        super().__init__(request, "started eligibility", eligibility_types)
+    def __init__(self, request, flow: models.EnrollmentFlow):
+        super().__init__(request, "started eligibility", flow)
 
 
 class ReturnedEligibilityEvent(EligibilityEvent):
     """Analytics event representing the end of an eligibility verification check."""
 
-    def __init__(self, request, eligibility_types, status, error=None):
-        super().__init__(request, "returned eligibility", eligibility_types)
+    def __init__(self, request, flow: models.EnrollmentFlow, status, error=None):
+        super().__init__(request, "returned eligibility", flow)
         status = str(status).lower()
         if status in ("error", "fail", "success"):
             self.update_event_properties(status=status, error=error)
-        if status == "success":
-            self.update_user_properties(eligibility_types=eligibility_types)
```

Comment:
Just noting for future reference that now we'll capture the enrollment flow as a user property for all `ReturnedEligibilityEvent`s, not just successful ones.

Reply:
This logic is (was) already present ... i.e. creating a new `EligibilityEvent` always updated the event/user properties with the flow (eligibility type in the past). 

I couldn't find a reason for this specific case of also updating it on success... it would have already been done by the user selecting their flow.

Good call out though.